hints/cygwin.sh

```diff
@@ -25,12 +25,21 @@ libswanted=`echo " $libswanted " | sed -e 's/ c / /g'`
 libswanted=`echo " $libswanted " | sed -e 's/ m / /g'`
 # - eliminate -lutil, symbols are all in libcygwin.a
 libswanted=`echo " $libswanted " | sed -e 's/ util / /g'`
+ignore_versioned_solibs='y'
+usenm='no'
+libc='/usr/lib/libcygwin.a'
+loclibpth=' '
+glibpth=' '
+plibpth=' '
+libpth=' '
+PATH='.:/usr/bin/'
```

Comment:
In general I don't think hints should just blindly define things, they should only define them if a value isn't already supplied.

Otherwise `./Configure -Dwhatever` where `whatever` is blindly overridden can get very confusing.

I'm guessing PATH is being set to influence the value set for LDLIBPTH when Makefile.SH is run, since ldlibpthname=PATH.  Of course this is lost if Makefile needs to be rebuilt.

Reply:
> In general I don't think hints should just blindly define things, they should only define them if a value isn't already supplied.

Several of these are also defined for other platforms this way. That said, adding «test -z "$foo" &&» like what's already done below may be